src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs

```diff
@@ -597,18 +597,31 @@ private void Run()
                 {
                     vVirtualMachineProfile.SecurityProfile = new SecurityProfile();
                 }
-                if (vVirtualMachineProfile.SecurityProfile.UefiSettings == null)
-                {
-                    vVirtualMachineProfile.SecurityProfile.UefiSettings = new UefiSettings();
-                }
                 vVirtualMachineProfile.SecurityProfile.SecurityType = this.SecurityType;
                 if (vVirtualMachineProfile.SecurityProfile.SecurityType.ToLower() == ConstantValues.TrustedLaunchSecurityType || vVirtualMachineProfile.SecurityProfile.SecurityType.ToLower() == ConstantValues.ConfidentialVMSecurityType)
                 {
+                    if (vVirtualMachineProfile.SecurityProfile.UefiSettings == null)
+                    {
+                        vVirtualMachineProfile.SecurityProfile.UefiSettings = new UefiSettings();
+                    }
                     vVirtualMachineProfile.SecurityProfile.UefiSettings.VTpmEnabled = vVirtualMachineProfile.SecurityProfile.UefiSettings.VTpmEnabled == null ? true : this.EnableVtpm;
                     vVirtualMachineProfile.SecurityProfile.UefiSettings.SecureBootEnabled = vVirtualMachineProfile.SecurityProfile.UefiSettings.SecureBootEnabled == null ? true : this.EnableSecureBoot;
                 }
             }
 
+            if (this.IsParameterBound(c => c.SecurityType) && this.SecurityType?.ToLower() == ConstantValues.StandardSecurityType)
```

Comment:
what is this.IsParameterBound(c => c.SecurityType) but not this.SecurityType?.ToLower() == ConstantValues.StandardSecurityType ?

Reply:
I think you're right that this can be removed and just use the above code but again, remove the 2nd conditional from the above if.